api/prometheus/v1/api_bench_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package v1
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/prometheus/common/model"
+)
+
+func generateData(timeseries, datapoints int) model.Matrix {
+	// Create the top-level matrix
+	m := make(model.Matrix, 0)
+
+	for i := 0; i < timeseries; i++ {
+		lset := map[model.LabelName]model.LabelValue{
+			model.MetricNameLabel: model.LabelValue("timeseries_" + strconv.Itoa(i)),
+		}
+		now := model.Now()
+		values := make([]model.SamplePair, datapoints)
+
+		for x := datapoints; x > 0; x-- {
+			values[x-1] = model.SamplePair{
+				Timestamp: now.Add(time.Second * -15 * time.Duration(x)), // Set the time back assuming a 15s interval
+				Value:     model.SampleValue(float64(x)),
+			}
+		}
+
+		ss := &model.SampleStream{
+			Metric: model.Metric(lset),
+			Values: values,
+		}
+
+		m = append(m, ss)
+	}
+	return m
+}
+
+func BenchmarkSerialization(b *testing.B) {
+	for _, timeseriesCount := range []int{10, 100, 1000} {
+		for _, datapointCount := range []int{10, 100, 1000} {
+			b.Run(fmt.Sprintf("%d/%d", timeseriesCount, datapointCount), func(b *testing.B) {
```

Comment:
Nested Run calls will do this for you.

Reply:
I'm not sure what you mean by this, are you suggesting to have 2 `b.Run` calls instead of one?

Reply:
Yes. It'll be a little simpler.

Reply:
I've split it, IMO its not simpler (since it requires more indentation and another function wrap).